### PR TITLE
Login to servers directly, don't rely on the client to do it

### PR DESCRIPTION
**Pull request by mipollex**
This goes along with [Skybunk-mobile 201](https://github.com/CGUC/skybunk-mobile/pull/201), supporting the ability to log in using the auth server.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,10 +13,10 @@
     "bcryptjs": "^2.4.3",
     "body-parser": "^1.19.0",
     "express": "^4.17.0",
-    "fetch": "^1.1.0",
     "http-errors": "^1.7.2",
     "lodash": "^4.17.11",
     "mongoose": "^5.5.11",
+    "node-fetch": "^2.6.0",
```
Comment (picklechips) [inline package.json:19]: What's the difference?

File: models/User.js [hunk 2]
```diff
@@ -139,22 +140,36 @@ UserSchema.methods.register = function(ticketNumber) {
...
+            fetch(`${server.url}/users/login`, {
+              method: 'POST',
+              headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json',
+              },
+              body: JSON.stringify({username, password}),
+            }).then(res => res.json()).then(jsonRes => {
+              if (jsonRes.err) {
+                reject({code: 404, message: jsonRes.err.message});
```
Comment (picklechips) [inline models/User.js:158]: :clap: Having the same error messages from the residence server will be nice

File: models/User.js [hunk 3]
```diff
@@ -139,22 +140,36 @@ UserSchema.methods.register = function(ticketNumber) {
...
       } else {
-        // Match password
-        bcrypt.compare(password, user.password, (err, isMatch) => {
-          if (isMatch && !err) {
-            user.populate('servers', (err) => {
-              if (err)
-                reject(err);
-              else
-                resolve(user);
+        Promise.all(user.servers.map(server => {
```
Comment (picklechips) [inline models/User.js:147]: Are we attempting to log in to every server that the user is registered for?

Then the response is an array of responses from every server?

File: package.json (hunk 1 above)
Comment (mipollex) [reply package.json:19]: `fetch` was actually the wrong library. It does something slightly different and I wanted to be consistent with mobile.
[fetch](https://www.npmjs.com/package/fetch) vs [node-fetch](https://www.npmjs.com/package/node-fetch)

File: models/User.js (hunk 3 above)
Comment (mipollex) [reply models/User.js:147]: yep. Once we're actually using more than 1 server per user we should reevaluate how we want failed login attempts to be handled.

This just logs into each server and creates an array of objects containing server name, server address, and auth token.

**Review by picklechips — APPROVED** [review]
_Might_ need a bit of rethinking once users can be registered for multiple servers, but looks good for now! :+1:

**Pull request merged**